src/domains/shielder/utils/useShielderFees.tsx

```diff
@@ -1,32 +1,161 @@
 import { erc20Token, nativeToken } from '@cardinal-cryptography/shielder-sdk';
 import { skipToken, useQuery } from '@tanstack/react-query';
-import { Address } from 'viem';
+import { useMemo } from 'react';
+import { z } from 'zod';
 
+import { useWallet } from 'src/domains/chains/components/WalletProvider.tsx';
 import { Token } from 'src/domains/chains/types/misc';
 import useChain from 'src/domains/chains/utils/useChain';
 import getQueryKey from 'src/domains/misc/utils/getQueryKey';
+import useEstimateAllowanceFee from 'src/domains/shielder/utils/useEstimateAllowanceFee';
 import useShielderClient from 'src/domains/shielder/utils/useShielderClient';
 
+export type FeeItem = {
+  type: 'network' | 'relayer' | 'allowance',
+  amount: bigint,
+  token: string,
+};
+
+export type FeeStructure = FeeItem[];
+
+const FeeEstimationSchema = z.object({
+  native_new_account_gas: z.string(),
+  native_deposit_gas: z.string(),
+  erc20_new_account_gas: z.string(),
+  erc20_deposit_gas: z.string(),
+  gas_price_native: z.string(),
+  update_timestamp: z.number(),
+});
+
+export type FeeEstimation = z.infer<typeof FeeEstimationSchema>;
+
 type Props = {
-  walletAddress: Address | undefined,
+  operation: 'shield' | 'send',
   token: Token,
+  amount?: bigint,
 };
 
-const useShielderFees = ({ walletAddress, token }: Props) => {
+const useShielderFees = ({ token, operation, amount }: Props) => {
   const chainConfig = useChain();
   const { data: shielderClient } = useShielderClient();
-  const { data } = useQuery(chainConfig ? {
-    queryKey: walletAddress ? getQueryKey.shielderFees(walletAddress, chainConfig.id.toString()) : [],
-    queryFn: !shielderClient ?
+  const { address: walletAddress } = useWallet();
+
+  const shieldingFeeEstimatorUrl = chainConfig?.shielderConfig?.shieldingFeeEstimatorUrl;
+  const tokenType = token.isNative ? 'native' : 'erc20';
+
+  const {
+    data: shieldingFees,
+    error: shieldingFeesError,
+    isLoading: areShieldingFeesLoading,
+  } = useQuery({
+    queryKey: chainConfig && shieldingFeeEstimatorUrl && operation === 'shield' ?
+      getQueryKey.shieldingFees(chainConfig.id.toString(), tokenType) : [],
+    queryFn: !shieldingFeeEstimatorUrl || operation !== 'shield' ?
+      skipToken :
+      async (): Promise<FeeEstimation> => {
+        const response = await fetch(shieldingFeeEstimatorUrl);
+
+        if (!response.ok) {
+          throw new Error(`Failed to fetch fees: ${response.status} ${response.statusText}`);
+        }
+
+        const rawData: unknown = await response.json();
+        return FeeEstimationSchema.parse(rawData);
```

Comment:
Just do 
```suggestion
        return FeeEstimationSchema.parse(await response.json());
```
and you don't need meaningless additional identifiers and types.